src/Sulu/Bundle/AdminBundle/Controller/ContentNavigationController.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/*
+ * This file is part of Sulu
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\AdminBundle\Controller;
+
+use FOS\RestBundle\Controller\Annotations\RouteResource;
+use FOS\RestBundle\Routing\ClassResourceInterface;
+use FOS\RestBundle\View\View;
+use FOS\RestBundle\View\ViewHandler;
+use FOS\RestBundle\View\ViewHandlerInterface;
+use Sulu\Bundle\AdminBundle\Navigation\ContentNavigationAliasNotFoundException;
+use Sulu\Bundle\AdminBundle\Navigation\ContentNavigationCollectorInterface;
+use Sulu\Component\Rest\Exception\RestException;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * This controller returns all the content navigation items for a given alias
+ * @RouteResource("content-navigation")
+ */
+class ContentNavigationController implements ClassResourceInterface
+{
+    /**
+     * @var ContentNavigationCollectorInterface
+     */
+    private $contentNavigationCollector;
+
+    /**
+     * @var ViewHandler
+     */
+    private $viewHandler;
+
+    public function __construct(
+        ContentNavigationCollectorInterface $contentNavigationCollector,
+        ViewHandlerInterface $viewHandler
+    ) {
+        $this->contentNavigationCollector = $contentNavigationCollector;
+        $this->viewHandler = $viewHandler;
+    }
+
+    /**
+     * Returns all the content navigation items for a given alias
+     * @param Request $request
+     * @return Response
+     */
+    public function cgetAction(Request $request)
+    {
+        try {
+            $alias = $request->get('alias');
+
+            if (!$alias) {
+                throw new RestException('The alias attribute is required to load the content navigation');
+            }
+
+            $options = $request->query->all();
+
+            $contentNavigationItems = $this->contentNavigationCollector->getNavigationItems($alias, $options);
+
+            $view = View::create($contentNavigationItems);
+        } catch (ContentNavigationAliasNotFoundException $exc) {
+            $restException = new RestException(
+                sprintf('The content navigation alias "%s" does not exist!', $exc->getAlias())
```

Comment:
That's not a runtime error IMHO. It can happen that somebody asks for a alias for which no service is registered. Just changing a parameter in a URL should not result in a 500. /cc @chirimoya 

But passing the previous exception seems to be valuable for me, I check if I can easily test that.